src/utils/raw.rs

```diff
@@ -52,20 +52,24 @@ pub fn if_not_empty<S: PartialEq<str>>(s: S) -> Option<S> {
     }
 }
 
-pub fn write_file(path: &Path, contents: &str) -> io::Result<()> {
+pub fn write_file_bytes(path: &Path, contents: &[u8]) -> io::Result<()> {
     let mut file = fs::OpenOptions::new()
         .write(true)
         .truncate(true)
         .create(true)
         .open(path)?;
 
-    io::Write::write_all(&mut file, contents.as_bytes())?;
+    io::Write::write_all(&mut file, contents)?;
 
     file.sync_data()?;
 
     Ok(())
 }
 
+pub fn write_file(path: &Path, contents: &str) -> io::Result<()> {
```

Comment:
I don't think this pays for itself vs write_file(path, contents.as_bytes())?;

Reply:
Fair enough, I'll sort out a refactor commit alongside this which pushes that up to the call sites.